### PR TITLE
CHANGES.md: Mention RSA key generation slowdown related changes

### LINKED ISSUE #14068: Missing CHANGES.md entry about change of default RSA key generation method to FIPS 186-4 B.3.6
The method is slower so there should be also some reasoning behind this method choice included.

### PR DESCRIPTION
**Pull request by t8m**
Fixes #14068

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

**Comment by t8m** [comment]
@kroeckx and @slontis please review

File: CHANGES.md [hunk 1]
```diff
@@ -35,7 +35,23 @@ OpenSSL 3.0
...
+   changed to FIPS 186-4 B.3.6 method (Generation of Probable Primes with
+   Conditions Based on Auxiliary Probable Primes). This method is slower
+   than the original method.
+
+   *Shane Lontis*
+
+ * Deprecated the BN_is_prime_ex() and BN_is_prime_fasttest_ex() functions.
+   They are replaced with BN_check_prime() function that avoids possible
+   misuse and always defaults to at least 64 rounds of Miller-Rabin
+   primality test. At least 64 rounds of MR test are now also used for
```
Comment (kroeckx) [inline CHANGES.md:48]: They are replaced with the BN_check_prime() function that avoids possible misuse and always uses at least 64 rounds of the Miller-Rabin primality test?

File: CHANGES.md [hunk 2]
```diff
@@ -35,7 +35,23 @@ OpenSSL 3.0
 
    *Tomáš Mráz*
 
- * Deprecate EVP_MD_CTX_set_update_fn() and EVP_MD_CTX_update_fn()
+ * Default key generation method for the regular 2-prime RSA keys was
```
Comment (kroeckx) [inline CHANGES.md:38]: The default?

File: CHANGES.md [hunk 3]
```diff
@@ -35,7 +35,23 @@ OpenSSL 3.0
...
+   changed to FIPS 186-4 B.3.6 method (Generation of Probable Primes with
```
Comment (kroeckx) [inline CHANGES.md:39]: the FIPS 186-4 B.3.6 method?

**Comment by t8m** [comment]
Fixup pushed

**Review by kroeckx — APPROVED** [review]

**Comment by kroeckx** [comment]
Why is the sanitizer build failing? Is it randomly failing?

**Comment by t8m** [comment]
> Why is the sanitizer build failing? Is it randomly failing?

Yes, #14070

File: CHANGES.md [hunk 4]
```diff
@@ -35,7 +35,23 @@ OpenSSL 3.0
...
+   than the original method.
+
+   *Shane Lontis*
+
+ * Deprecated the BN_is_prime_ex() and BN_is_prime_fasttest_ex() functions.
+   They are replaced with the BN_check_prime() function that avoids possible
+   misuse and always uses at least 64 rounds of the Miller-Rabin
+   primality test. At least 64 rounds of MR test are now also used for
+   all prime generation including the RSA key generation which increases
+   the key generation time especially for larger keys.
```
Comment (h-vetinari) [inline CHANGES.md:50]: Personal preference would also be: ~the~ RSA key generation
```suggestion
   all prime generation, including RSA key generation, which increases
   key generation time, especially for larger keys.
```

Comment (paulidale) [reply CHANGES.md:50]: I'd suggest breaking it into two sentences instead and avoiding the acronym:

> At least 64 rounds of the Miller-Rabin test are now also used for all prime generation, including RSA key generation.  This increases key generation time, especially for larger keys.

**Review by paulidale — COMMENTED** [review]
I mostly like @h-vetinari 's suggestion, although I suggest wording it slightly differently.

**Comment by t8m** [comment]
Changed according to @paulidale suggestion. Please re-approve.

**Review by paulidale — APPROVED** [review]
Great!

**Review by kroeckx — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master. Thank you for the reviews.

**Pull request closed without merging**